lib/index.js

```diff
@@ -469,6 +471,74 @@ class AuthenticationPlugin {
     return this.getUsersRepository().delete(user._id, {refresh: 'wait_for'});
   }
 
+  /**
+   * Search users by ES query concerning exclusively their username
+   *
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (undefined), size (undefined)
+   *
+   * @returns {Promise.<{ hits, total }>}
+   */
+  async search (searchBody, { from, size } = {}) {
+    debug('exists(searchBody = %j)', searchBody);
+
+    let sort;
+    let query;
+    let isRequiredFieldsInBody = false;
+    const forbiddenFields = extractESMappingFields(storageMapping.users);
+    const forbiddenKeywords = [
+      'multi_match',
+    ]; // @todo Keep up to date this blacklist to prevent queries on forbiddenFields
+
+    const sanitizeSearchBody = (key, value) => {
+      if (key === 'username') {
+        isRequiredFieldsInBody = true;
+        return { key: '_id' };
+      }
+      else if (value === 'username') {
+        isRequiredFieldsInBody = true;
+        return { value: '_id' };
+      }
+      else if (forbiddenKeywords.includes(key)) {
+        throw new this.context.errors.BadRequestError(`The "${key}" keyword is not allowed in this search query for security concerns.`);
+      }
+      else if (forbiddenFields.some((property) => property === key || property === value)) {
+        throw new this.context.errors.ForbiddenError(`Forbidden field "${key}". Only the "username" or "kuid" fields are sortable and only the first is also searchable.`);
```

Comment:
`ForbiddenError` are used when you try to access a resource you are not authorized to.
IMHO  a `BadRequestError` is more appropriate here

Reply:
@scottinet scored a point in this [comment](https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/pull/76#discussion_r634205973) about `ForbiddenError`, I think this should be left like it is.

Reply:
It's tricky. After giving it some thoughts, I believe that both errors can apply:
* Forbidden: well, as you said it @Aschen, a user tried to access to unauthorized resources, so this error applies
* BadRequest: applies to requests that are ill-formed. If the documentation states that only the `username` field is searchable, then this can, too, be considered as an ill-formed query.

I think there is no wrong answer here.